babel.config.js

```diff
@@ -25,99 +25,394 @@
+    [localPlugin('transform-log-methods'), {replaceCallArguments: false}],
+    localPlugin('transform-parenthesize-expression'),
+    localPlugin('is_minified-constant-transformer'),
+    localPlugin('transform-amp-extension-call'),
+    localPlugin('transform-html-template'),
+    localPlugin('transform-version-call'),
+    localPlugin('transform-simple-array-destructure'),
+    replacePlugin,
+  ];
+  if (argv.single_pass) {
+    preClosurePlugins.push(localPlugin('transform-amp-asserts'));
+  }
+  if (argv.esm) {
+    preClosurePlugins.push(
+      defaultFilterImportsPlugin,
+      localPlugin('transform-function-declarations')
+    );
+  }
+  const isCheckTypes = argv._.includes('check-types');
+  if (isCheckTypes) {
+    preClosurePlugins.push(localPlugin('transform-simple-object-destructure'));
+  } else {
+    preClosurePlugins.push(localPlugin('transform-json-configuration'));
+  }
+  if (!argv.fortesting && !isCheckTypes) {
+    preClosurePlugins.push(
+      [localPlugin('amp-mode-transformer'), {isEsmBuild: argv.esm}],
+      localPlugin('is_dev-constant-transformer')
+    );
   }
+  const babelPresetEnvOptions = argv.esm
+    ? {bugfixes: true, modules: false, targets: {esmodules: true}}
+    : {
+        bugfixes: true,
+        loose: true,
+        modules: false,
+        targets: {'browsers': ['Last 2 versions']},
+      };
+  const preClosurePresets = [['@babel/preset-env', babelPresetEnvOptions]];
+  const preClosureConfig = {
+    compact: false,
+    plugins: preClosurePlugins,
+    presets: preClosurePresets,
+    retainLines: true,
+  };
+  if (argv.esm) {
+    preClosureConfig.sourceType = 'module';
+  }
+  return preClosureConfig;
+}
 
-  return [
-    leadingComments,
-    reactConstantElements,
-    transformClasses,
-    transformJSX,
-  ];
-};
+/**
+ * Gets the config for post-closure babel transforms run during `gulp dist`.
+ *
+ * @return {!Object}
+ */
+function getPostClosureConfig() {
+  const postClosurePlugins = argv.esm
+    ? [
+        localPlugin('transform-minified-comments'),
+        localPlugin('transform-function-declarations'),
+        localPlugin('transform-stringish-literals'),
+      ]
+    : [];
+  return {
+    inputSourceMap: false,
+    plugins: postClosurePlugins,
+    retainLines: false,
+    sourceMaps: true,
+    sourceType: 'module',
+  };
+}
 
-const presets = (esm) => {
-  if (esm) {
-    return [
-      [
-        '@babel/preset-env',
-        {
-          'modules': false,
-          'targets': targets(esm),
-          'bugfixes': true,
-        },
-      ],
-    ];
-  }
-
-  return [
-    [
-      '@babel/preset-env',
-      {
-        'modules': isClosureCompiler ? false : 'commonjs',
-        'loose': true,
-        'targets': targets(esm),
-      },
-    ],
+/**
+ * Gets the config for babel transforms run during `gulp dist --single_pass`.
+ *
+ * @return {!Object}
+ */
+function getSinglePassConfig() {
+  const singlePassPlugins = [localPlugin('transform-prune-namespace')];
```

Comment:
Is this pre or post closure? A defined structure would make this easier to answer after a couple of days away.